Use the same df flags on macOS as on FreeBSD and add the -P flag to
suppress printing the inode columns. In recent versions of macOS df
prints the inode columns by default but xdiskusage relies on the output
of df having a certain number of columns and does not expect the inode
columns to be present.
https://sourceforge.net/p/xdiskusage/bugs/26/
--- xdiskusage.C.orig	2020-06-18 18:15:39.000000000 -0500
+++ xdiskusage.C	2020-06-18 18:16:00.000000000 -0500
@@ -28,8 +28,8 @@
 #elif defined(__bsdi__)
 # define DF_COMMAND "df"
 # define DU_COMMAND "du -x"
-#elif defined(__FreeBSD__)
-# define DF_COMMAND "df -k -t noprocfs,devfs,fdescfs"
+#elif defined(__FreeBSD__) || defined(__APPLE__)
+# define DF_COMMAND "df -P -k -t noprocfs,devfs,fdescfs"
 # define DU_COMMAND "du -kx"
 #elif defined(SVR4) || defined(__sun)
 # define DF_COMMAND "/bin/df -k"
